### PR TITLE
feat(cli): add --forbid-focused flag

### PR DESCRIPTION
**Pull request by lkipke**
# Change Summary

This allows a consumer to pass `--forbid-focused` to the `roca` cli, which will then cause the test runner to exit with error if it detects a focused test case, i.e. it emulates the [Mocha cli `--forbid-only`](https://mochajs.org/#command-line-usage). This is useful to be able to configure CI/CD to avoid accidentally checking in focused tests/suites.

### CODE REVIEW

File: resources/roca_main.brs [hunk 1]
```diff
@@ -48,12 +47,12 @@ sub main()
...
 
-        path = ["pkg:", basePath, file].join("/")
-        suite = _brs_.runInScope(path, args)
+        filePath = [basePath, file].join("/")
+        suite = _brs_.runInScope(["pkg:", filePath].join("/"), args)
 
         ' If brs returned invalid for runInScope, that means the suite threw an exception, so we should bail.
         if suite = invalid then
-            tap.bail("Error occurred in " + [basePath, file].join("/"))
+            tap.bail("Error running tests: Runtime exception occurred in " + filePath)
```
Comment (lkipke) [inline resources/roca_main.brs:55]: This is unrelated; just wanted to clean up the error message.

File: src/index.js [hunk 2]
```diff
@@ -25,19 +27,35 @@ async function runTest(files, reporter, requireFilePath) {
...
         });
+
         reporterStream.end();
+
+        if (forbidFocused && returnVals.length > 0) {
+            // iterate through return values and see if there are focused cases
+            for (output of returnVals) {
+                let focusedFiles = output.getValue().get("fileswithfocusedcases").getElements();
+                if (focusedFiles.length > 0) {
+                    let formattedList = focusedFiles.map((brsStringPath) => `    ${brsStringPath.value}`).join("/n");
```
Comment (alimnios72) [inline src/index.js:42]: does the formatted list work if you use a `\t` instead of the white spaced? also, shouldn't  the last part be `join(\/n");`

File: resources/roca_main.brs [hunk 3]
```diff
@@ -1,38 +1,37 @@
...
+            end if
+
+            rootSuites.push(subSuite)
+        end for
     end for
 
     tap = tap()
     tap.version()
 
+    focusedCasesDetected = filesWithFocusedCases.count() > 0
```
Comment (kirkyoder) [inline resources/roca_main.brs:32]: Mostly a nit, but rather than doing this boolean evaluation, here, consider just holding onto the value of `count()`. That we don't need to reevaluate it, below, on line 34:
```brightscript
numFocusedFiles = filesWithFocusedCases.count()
if numFocusedCases > 0 then
    tap.plan(numFocusedFiles)
else
    tap.plan(rootSuites.count())
end if
```

File: src/index.js [hunk 4]
```diff
@@ -3,14 +3,16 @@ const glob = require('glob');
 const path = require('path');
 const util = require('util');
 const TapMochaReporter = require('tap-mocha-reporter');
+const c = require("ansi-colors");
```
Comment (kirkyoder) [inline src/index.js:6]: 👶 nit: match the existing style convention for single vs double quotes?

File: src/index.js (hunk 2 above)
Comment (kirkyoder) [inline src/index.js:42]: I think you've got the wrong slash for this escape sequence 🤓 
```suggestion
                    let formattedList = focusedFiles.map((brsStringPath) => `    ${brsStringPath.value}`).join("\n");
```

Comment (kirkyoder) [reply src/index.js:42]: +1 to using `\t`

File: src/index.js [hunk 5]
```diff
@@ -25,19 +27,35 @@ async function runTest(files, reporter, requireFilePath) {
...
         reporterStream.end();
+
+        if (forbidFocused && returnVals.length > 0) {
+            // iterate through return values and see if there are focused cases
+            for (output of returnVals) {
+                let focusedFiles = output.getValue().get("fileswithfocusedcases").getElements();
+                if (focusedFiles.length > 0) {
+                    let formattedList = focusedFiles.map((brsStringPath) => `    ${brsStringPath.value}`).join("/n");
+                    console.error(c.red(`Error: used command line arg ${c.cyan("--forbid-focused")} but found focused tests in these files:`));
+                    console.error(formattedList);
```
Comment (kirkyoder) [inline src/index.js:44]: Just a 💭 ... I wonder if it'd be more appropriate to collapse these into a single `error()` statement for better readability in the Dev Console? 🤷‍♂️ 🐳

File: README.md [hunk 6]
```diff
@@ -134,9 +135,17 @@ end function
 ### Output & CI Support
 Roca exclusively reports its state via the [Test Anything Protocol](http://testanything.org/), and defaults to a Mocha-like "spec" output.  Failed tests cause the `roca` CLI to return a non-zero exit code, which allows most continuous integration systems to automatically detect pass/fail states.
 
-Other output formats are available!  See `roca --help` for more details.
```
[the remark this answers is not in the document]
Comment (lkipke) [reply README.md:137]: It is! I just thought the table i added would be a clearer way of communicating this

File: src/index.js (hunk 4 above)
Comment (lkipke) [reply src/index.js:6]: ah true, this file is split anyway so maybe i'll just convert them all

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (sjbarag) [reply README.md:137]: I think it's covered by `-R`/`--reporter` 😃

File: resources/roca_main.brs (hunk 3 above)
Comment (lkipke) [reply resources/roca_main.brs]: We use `focusedCasesDetected` in 3 other places later on, and it feels a little overkill to have so many variables (thus adding more "sources of truth") when `.count()` exists.

**Comment by sjbarag** [comment]
> @kirkyoder: Looks great! I wish we had unit tests... _meta_, I know. 😄

🕵️ Weird, someone just opened #45 to add tests to this project
